app/src/main/java/app/revanced/integrations/youtube/requests/Requester.java

```diff
@@ -24,7 +24,10 @@ public static HttpURLConnection getConnectionFromCompiledRoute(String apiUrl, Ro
         String url = apiUrl + route.getCompiledRoute();
         HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
         connection.setRequestMethod(route.getMethod().name());
-        connection.setRequestProperty("User-Agent", System.getProperty("http.agent") + "; ReVanced/" + Utils.getVersionName());
+        String agentString = System.getProperty("http.agent")
+                + "; ReVanced/" + Utils.getAppVersionName()
+                + "-" + Utils.getPatchesReleaseVersion();
```

Comment:
The user agent is built upon a specification: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent.

I think a different approach than sending information to the server is better but a little more complex.
The announcement could ship an executable with a hash. The app would dynamically execute it and evaluate if the announcement should be shown. This approach gives us a lot of freedom for the future to check for specific conditions such as the patches or app version or other kinds of situations in which we may or may not want the user to show an announcement. This approach is better than assuming that certain information will be sufficient to determine if an announcement should be shown as proposed in this PR, where we assume the patch version will be sufficient.

This idea is similar to how Google operates DroidGuard, with the difference that it operates obfuscated, leading to the following concern: Because the client would download executables remotely, this could pose a security risk. The executable must be signed, and the signer must be trusted before the executable is run. In our case, the user must trust us as the signer. Once they trust us, they can check the executable's sign before running it in a trusted manner. The executable source has to be open as well and auditable. If the signer is untrusted, the executable will not be downloaded/run, and every announcement could be shown.

In terms of bandwidth, the announcement could come with an executable hash. The client can download the executable from a specific endpoint if the hash is present. If it is not present, no evaluation is necessary for that announcement.

This idea involves a lot of implementation and more thought and decisions before being realized which is why for now, the change in user agent should be done in a different PR in consideration to this idea.